packages/app/src/debug/DebugSpec.vue

```diff
@@ -0,0 +1,122 @@
+        class="w-full flex flex-row items-center py-12px bg-gray-50 border-b-1px border-b-gray-100 rounded-t"
+      >
+        <div
+          data-cy="spec-contents"
+          class="w-full flex flex-row px-16px items-center"
+        >
+          <div
+            data-cy="spec-path"
+            class="flex-grow non-italic text-base truncate"
+          >
+            <span
+              class="font-normal text-gray-600"
+            >
+              {{ specData.path }}
+            </span>
+            <span
+              class="font-medium text-gray-900"
+            >
+              {{ specData.fileName }}
+            </span>
+            <span
+              class="font-normal text-gray-600"
+            >
+              {{ specData.extension }}
+            </span>
+          </div>
+          <div
+            class="ml-10px"
+          >
+            <Button
+              data-cy="run-failures"
+              variant="white"
+              class="inline-flex gap-x-10px whitespace-nowrap justify-center items-center isolate"
+              :disabled="isDisabled"
+              :to="{ path: '/specs/runner', query: { file: (specData.path + specData.fileName + specData.extension).replace(/\\/g, '/') } }"
+            >
+              <template #prefix>
+                <IconActionRefresh
+                  data-cy="icon-refresh"
+                  stroke-color="indigo-500"
+                />
+              </template>
+              Run Failures
+            </Button>
+          </div>
+        </div>
+      </div>
+      <ul
+        v-for="test in specData.failedTests"
+        :key="`test-${test.id}`"
+        data-cy="test-group"
+        class="w-full flex flex-col flex-start h-12 items-start justify-center pl-16px border-b-gray-100 border-b-1px"
+      >
+        <DebugFailedTest
+          :failed-test-result="test"
+          :data-cy="`test-${test.id}`"
+        />
+      </ul>
+    </div>
+  </div>
+</template>
+<script lang="ts" setup>
+export interface Spec {
+  id: string
+  path: string
+}
+
+export interface TestResults {
+  id: string
+  titleParts: string[]
+}
+
+import { computed } from 'vue'
+import { IconActionRefresh } from '@cypress-design/vue-icon'
+import DebugFailedTest from './DebugFailedTest.vue'
+import Button from '@packages/frontend-shared/src/components/Button.vue'
+
+const props = defineProps<{
+  spec: Spec
+  testResults: TestResults[]
+  isDisabled?: boolean
+}>()
+
+/**
+ * @param relativePath: The relative path of a failed test e.g. cypress/test/auth.spec.ts
+ * We split the relative path into 3 pieces and return an object containing this split sections
+ * so we can correctly style the relative path in the debug spec header. For example:
+ * path: cypress/tests/ ; fileName: auth ; extension: .spec.ts
+ */
+const splitHeader = (relativePath: string) => {
+  const before = relativePath.split('/')
+  const specName = before.slice(-1)[0]
+  const splitSpecName = specName.split('.')
```

Comment:
What about a spec filename like `my.component.spec.cy.js`? I think this logic would only end up highlighting `my` when it should cover `my.component.spec`. Our GQL schema exposes `Spec.fileName` and `Spec.specFileExtension` fields that are already pre-parsed, may want to see if those can be reused and just accept those prepared values as props

Reply:
@warrensplayer Would it be possible to pass in `Spec.fileName` and `Spec.specFileExtension` instead of  `path` to this component when using it in the parent component? 
My understanding is that the GQL query will be in the parent component, therefore if this is something that could be done, that it would overcome several of these edges cases and we could simplify the `splitHeader` method

Reply:
I am a bit out of the loop for this project - does this data come straight from the Cloud or going via our data-context?

We've got logic to get all the parts of a file here: https://github.com/cypress-io/cypress/blob/develop/packages/data-context/src/sources/ProjectDataSource.ts#L94-L104 we use this for the Specs List and some other places. I think, if possible, we should do all this preprocessing on the server and then serve it up using the same interface we use for the rest of App https://github.com/cypress-io/cypress/blob/develop/packages/graphql/src/schemaTypes/objectTypes/gql-Spec.ts

Not sure if we need to do this here or later on, just some ideas. Good to reuse battle tested code where we can.

Reply:
@amehta265 Yes, I see the logic that Lachlan is pointing to above and we can make sure the new GraphQL fields take advantage of that same parsing logic.  You can remove that parsing logic and update the props of the component to take something like what Mike is talking about on the Specs List page: 
```
{
  path: '/src/main/components',
  fileName: 'MyComponet',
  fileExtension: '.cy.ts'
}
```